### DISCUSSION TITLE
How to test this locally?

### DISCUSSION THREAD

**Discussion by robfuncken** [caching-tools/next-shared-cache#443] [Q&A] [answered]
Hi all, 

We're deploying a Next.js app to Azure App Services and want to make use of horizontal scaling. Thats why i'd like to use this package. 
But I can't seem to get it to work locally, both with version 0.6 and with 1.0.7. When I do a npm run build && npm run start with the [official nextjs example](https://github.com/vercel/next.js/tree/canary/examples/cache-handler-redis) (with version 0.6), or with the examples provided and using 1.0.7, there is no cache written to the Redis store. 
I've tried with a local redis and with a deployed redis instance. But no luck.

Is it because of the fact that I don't run multiple instances locally? 

How can I verify my cachehandler is working correctly? I've turned on the ENV NEXT_PRIVATE_DEBUG_CACHE=1

**Comment by better-salmon** [comment, marked as answer]
@robfuncken  hello! I'm so sorry it took so long to answer.

Let's assume that version before 1.0.0 is obsolete, and we will discuss the latest version. And I must update the Next.js example to the latest version.

Here are the required steps for local debugging:

1. Install and run the Redis server locally.
2. Install the `@neshca/cache-handler` and the `redis` using your package manager.
3. Create a `cache-handler.mjs` file near the `next.config.(m)js` file and paste there the code from the [Redis Handler example](https://caching-tools.github.io/next-shared-cache/redis).
4. Optionally change import from `@neshca/cache-handler/redis-stack` to `@neshca/cache-handler/redis-strings` if your are going to use Redis server prior to version 7. `@neshca/cache-handler/redis-strings` requires Redis server 4.0.0 and newer.
5. Add `cacheHandler` option to your `next.config.(m)js` file. See the [installation guide](https://caching-tools.github.io/next-shared-cache/installation).
6. Add `NEXT_PRIVATE_DEBUG_CACHE=1` to your `.env.local`.
7. Build your application with `npm run build` command.
8. Start the application with `npm run start`.
9. Open one of the routes with `revalidation` or with fetch inside the server component. (or route with `getStaticProps` if you are using the Pages router).

In the console of your next server you should see something like this:

```
using custom cache handler @neshca/cache-handler with 1 Handlers
get from "redis-stack" ba7bea0a3991dc2cbf8e068da71c2a2aa8c0e6f902fe695c1badf1e994acc43a false
set to external cache store 8f058e8d96c4010b07abddd0f9f26ebd9ed71e2ccfb18eca348fc57a6e86f986
```

If that is not the case, make sure that you, the Redis client, have managed to establish the connection with the Redis server.

I hope that will help you!
